### PR TITLE
AUTH-123: Automated testing of SSO for APS and ACS

### PR DESCRIPTION
**Pull request by kconkas**
Added token authentication based SSO tests. 

**Important:** do not merge this to master until AUTH-124 has been merged to it.

### CODE REVIEW

File: test/postman/dbp-test-collection.json [hunk 1]
```diff
@@ -863,8 +963,7 @@
 					"raw": ""
 				},
 				"url": {
-					"raw": "http://{{url}}/actuator/routes",
```
Comment (svidrascu) [inline test/postman/dbp-test-collection.json]: Why did we remove http here? will this work like this, in ci we are only giving the elb name not the protocol.

File: test/postman/dbp-test-collection.json [hunk 2]
```diff
@@ -812,41 +894,59 @@
...
-			"name": "auth gateway",
+			"name": "Auth Gateway routes",
 			"event": [
 				{
 					"listen": "test",
 					"script": {
+						"id": "db57447f-44f7-4e84-bbc2-7ed8ebf53c4b",
+						"type": "text/javascript",
 						"exec": [
-							"pm.globals.get(\"url\");",
```
Comment (svidrascu) [inline test/postman/dbp-test-collection.json]: Wasnt this check done with token auth?

File: test/postman/dbp-test-collection.json [hunk 3]
```diff
@@ -898,5 +997,84 @@
...
+			"id": "6dbd49bb-9839-44ca-9e1e-8c06b256d77d",
+			"key": "identityServiceUrl",
+			"value": "{{url}}/auth",
+			"type": "string",
+			"description": ""
+		},
+		{
+			"id": "c66d9815-e0e2-4fb0-8ec3-5650d14c136b",
+			"key": "identityServiceRealm",
+			"value": "springboot",
```
Comment (svidrascu) [inline test/postman/dbp-test-collection.json]: Shouldn't this be grabbed from global with {{IdentityServiceRealm}}

File: test/postman/dbp-test-collection.json (hunk 1 above)
Comment (kconkas) [reply test/postman/dbp-test-collection.json]: The protocol was removed in order to enable testing with either http or https. As for CI, there is also an accompanying change to cater for this https://git.alfresco.com/platform-services/bamboo-build-dbp/merge_requests/16.

File: test/postman/dbp-test-collection.json (hunk 2 above)
Comment (kconkas) [reply test/postman/dbp-test-collection.json]: This line was redundant, simply obtaining the value of global variable called "url" and subsequently doing nothing with it.

File: test/postman/dbp-test-collection.json (hunk 1 above)
Comment (svidrascu) [reply test/postman/dbp-test-collection.json]: Ok, thanks for the explanation.

File: test/postman/dbp-test-collection.json (hunk 3 above)
Comment (kconkas) [reply test/postman/dbp-test-collection.json]: I'm not sure I quite understand this comment: this _is_ a global variable.

Comment (svidrascu) [reply test/postman/dbp-test-collection.json]: Is this the case for the url variable also? is that a collection variable or global?
If its still global we should turn that to collection also.

**Comment by kconkas** [comment]
@svidrascu our discussion about variable scope flagged the fact there was one global variable the collection used to set to pass the authorization token. I've tidied this now to be an environment variable.

**Pull request merged**